lib/matplotlib/backends/wx_compat.py

```diff
@@ -0,0 +1,162 @@
+    fontnames = {'Sans': wx.FONTFAMILY_SWISS,
+                 'Roman': wx.FONTFAMILY_ROMAN,
+                 'Script': wx.FONTFAMILY_SCRIPT,
+                 'Decorative': wx.FONTFAMILY_DECORATIVE,
+                 'Modern': wx.FONTFAMILY_MODERN,
+                 'Courier': wx.FONTFAMILY_MODERN,
+                 'courier': wx.FONTFAMILY_MODERN}
+
+    dashd_wx = {'solid': wx.PENSTYLE_SOLID,
+                'dashed': wx.PENSTYLE_SHORT_DASH,
+                'dashdot': wx.PENSTYLE_DOT_DASH,
+                'dotted': wx.PENSTYLE_DOT}
+
+    # functions changes
+    BitmapFromBuffer = wx.Bitmap.FromBufferRGBA
+    EmptyBitmap = wx.Bitmap
+    EmptyImage = wx.Image
+    Cursor = wx.Cursor
+    EventLoop = wx.GUIEventLoop
+    NamedColour = wx.Colour
+    StockCursor = wx.Cursor
+
+else:
+    # define all the wxPython classic stuff
+
+    # font styles, families and weight
+    fontweights = {
+        100: wx.LIGHT,
+        200: wx.LIGHT,
+        300: wx.LIGHT,
+        400: wx.NORMAL,
+        500: wx.NORMAL,
+        600: wx.NORMAL,
+        700: wx.BOLD,
+        800: wx.BOLD,
+        900: wx.BOLD,
+        'ultralight': wx.LIGHT,
+        'light': wx.LIGHT,
+        'normal': wx.NORMAL,
+        'medium': wx.NORMAL,
+        'semibold': wx.NORMAL,
+        'bold': wx.BOLD,
+        'heavy': wx.BOLD,
+        'ultrabold': wx.BOLD,
+        'black': wx.BOLD
+    }
+    fontangles = {
+        'italic': wx.ITALIC,
+        'normal': wx.NORMAL,
+        'oblique': wx.SLANT}
+
+    # wxPython allows for portable font styles, choosing them appropriately
+    # for the target platform. Map some standard font names to the portable
+    # styles
+    # QUESTION: Is it be wise to agree standard fontnames across all backends?
+    fontnames = {'Sans': wx.SWISS,
+                 'Roman': wx.ROMAN,
+                 'Script': wx.SCRIPT,
+                 'Decorative': wx.DECORATIVE,
+                 'Modern': wx.MODERN,
+                 'Courier': wx.MODERN,
+                 'courier': wx.MODERN}
+
+    dashd_wx = {'solid': wx.SOLID,
+                'dashed': wx.SHORT_DASH,
+                'dashdot': wx.DOT_DASH,
+                'dotted': wx.DOT}
+
+    # functions changes
+    BitmapFromBuffer = wx.BitmapFromBufferRGBA
+    EmptyBitmap = wx.EmptyBitmap
+    EmptyImage = wx.EmptyImage
+    Cursor = wx.StockCursor
+    EventLoop = wx.EventLoop
+    NamedColour = wx.NamedColour
+    StockCursor = wx.StockCursor
+
+
+def AddTool(parent, wx_ids, text, bmp, tooltip_text):
+    if is_phoenix:
+        if text in ['Pan', 'Zoom']:
+            kind = wx.ITEM_CHECK
+        else:
+            kind = wx.ITEM_NORMAL
+        parent.AddTool(wx_ids[text], label=text,
+                       bitmap=bmp,
+                       bmpDisabled=wx.NullBitmap,
+                       shortHelpString=text,
+                       longHelpString=tooltip_text,
+                       kind=kind)
+    else:
```

Comment:
one thing I worry about with this logic is that it is only really useful for internal use. It is a compatibility layer that is specific to matplotlib's navigation tools. It doesn't really help anybody else. Perhaps this function name should get an underscore?